File: .sling/libs/sushi/eqela/jk-core/0.25.0/jk.env.lua
jk = jk or {}

jk.env = jk.env or {}

jk.env.TemporaryDirectory = {}
jk.env.TemporaryDirectory.__index = jk.env.TemporaryDirectory
_vm:set_metatable(jk.env.TemporaryDirectory, {})

function jk.env.TemporaryDirectory._create()
	local v = _vm:set_metatable({}, jk.env.TemporaryDirectory)
	return v
end

function jk.env.TemporaryDirectory:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.env.TemporaryDirectory'
	self['_isType.jk.env.TemporaryDirectory'] = true
end

function jk.env.TemporaryDirectory:_construct0()
	jk.env.TemporaryDirectory._init(self)
	return self
end

function jk.env.TemporaryDirectory:create()
	do return _g.jk.env.TemporaryDirectory:forDirectory(nil) end
end

function jk.env.TemporaryDirectory:forDirectory(dir)
	local ff = _g.jk.env.TemporaryFile:forDirectory(dir, nil)
	if not (ff ~= nil) then
		do return nil end
	end
	do ff:remove() end
	do ff:createDirectoryRecursive() end
	if not ff:isDirectory() then
		do return nil end
	end
	do return ff end
end

jk.env.ExternalCommand = {}
jk.env.ExternalCommand.__index = jk.env.ExternalCommand
_vm:set_metatable(jk.env.ExternalCommand, {})

function jk.env.ExternalCommand._create()
	local v = _vm:set_metatable({}, jk.env.ExternalCommand)
	return v
end

function jk.env.ExternalCommand:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.env.ExternalCommand'
	self['_isType.jk.env.ExternalCommand'] = true
end

function jk.env.ExternalCommand:_construct0()
	jk.env.ExternalCommand._init(self)
	return self
end

function jk.env.ExternalCommand:findInPath(command)
	if not (command ~= nil) then
		do return nil end
	end
	do
		local path = _g.jk.env.EnvironmentVariable:get("PATH")
		if not _g.jk.lang.String:isNotEmpty(path) then
			do return nil end
		end
		do
			local separator = 58
			if _g.jk.os.OS:isWindows() then
				separator = 59
			end
			do
				local array = _g.jk.lang.String:split(path, separator, 0)
				if array ~= nil then
					local n = 0
					local m = _g.jk.lang.Vector:getSize(array)
					do
						n = 0
						while n < m do
							local dir = array[n + 1]
							if dir ~= nil then
								local pp = _g.jk.fs.File:forPath(dir):rawEntry(command):asExecutable()
								if pp:isFile() then
									do return pp end
								end
							end
							do n = n + 1 end
						end
					end
				end
				do return nil end
			end
		end
	end
end

jk.env.CommonPath = {}
jk.env.CommonPath.__index = jk.env.CommonPath
_vm:set_metatable(jk.env.CommonPath, {})

function jk.env.CommonPath._create()
	local v = _vm:set_metatable({}, jk.env.CommonPath)
	return v
end

function jk.env.CommonPath:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.env.CommonPath'
	self['_isType.jk.env.CommonPath'] = true
end

function jk.env.CommonPath:_construct0()
	jk.env.CommonPath._init(self)
	return self
end

function jk.env.CommonPath:getTemporaryDirectory()
	if _g.jk.os.OS:isLinux() or _g.jk.os.OS:isMacOS() then
		do return _g.jk.fs.File:forPath("/tmp") end
	end
	_io:write_to_stdout("[jk.env.CommonPath.getTemporaryDirectory] (CommonPath.sling:45:3): Not implemented" .. "\n")
	do return _g.jk.fs.FileInvalid._construct0(_g.jk.fs.FileInvalid._create()) end
end

function jk.env.CommonPath:getHomeDirectory()
	local dirname = _g.jk.env.EnvironmentVariable:get("HOME")
	if _g.jk.lang.String:isNotEmpty(dirname) then
		do return _g.jk.fs.File:forPath(dirname) end
	end
	do return nil end
end

function jk.env.CommonPath:getAppDirectory()
	local program = _vm:get_program_path()
	if not _g.jk.lang.String:isNotEmpty(program) then
		do return nil end
	end
	do
		local pf = _g.jk.fs.File:forPath(program)
		if not (pf ~= nil) then
			do return nil end
		end
		do return pf:getParent() end
	end
end

function jk.env.CommonPath:getCodeFileForObject(oo)
	if not (oo ~= nil) then
		do return nil end
	end
	do return nil end
end

function jk.env.CommonPath:getProgramFilesDirectory()
	do return nil end
end

function jk.env.CommonPath:getProgramFilesX86Directory()
	do return nil end
end

function jk.env.CommonPath:getApplicationsDirectory()
	if _g.jk.os.OS:isMacOS() then
		local v = _g.jk.fs.File:forPath("/Applications")
		if v:isDirectory() == false then
			v = nil
		end
		do return v end
	end
	do return nil end
end

jk.env.EnvironmentVariable = {}
jk.env.EnvironmentVariable.__index = jk.env.EnvironmentVariable
_vm:set_metatable(jk.env.EnvironmentVariable, {})

function jk.env.EnvironmentVariable._create()
	local v = _vm:set_metatable({}, jk.env.EnvironmentVariable)
	return v
end

function jk.env.EnvironmentVariable:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.env.EnvironmentVariable'
	self['_isType.jk.env.EnvironmentVariable'] = true
end

function jk.env.EnvironmentVariable:_construct0()
	jk.env.EnvironmentVariable._init(self)
	return self
end

function jk.env.EnvironmentVariable:getAll()
	local v = {}
	local envs = _os:get_all_environment_variables()
	if envs ~= nil then
		if envs ~= nil then
			local n = 0
			local m = _g.jk.lang.Vector:getSize(envs)
			do
				n = 0
				while n < m do
					local env = envs[n + 1]
					if env ~= nil then
						local comps = _g.jk.lang.String:split(env, 61, 2)
						local key = _g.jk.lang.Vector:get(comps, 0)
						local val = _g.jk.lang.Vector:get(comps, 1)
						if not (key ~= nil) then
							goto _continue1
						end
						if not (val ~= nil) then
							val = ""
						end
						do _g.jk.lang.Map:set(v, key, val) end
					end
					::_continue1::
					do n = n + 1 end
				end
			end
		end
	end
	do return v end
end

function jk.env.EnvironmentVariable:get(key)
	if not (key ~= nil) then
		do return nil end
	end
	do
		local v = nil
		do v = _os:get_environment_variable(key) end
		do return v end
	end
end

function jk.env.EnvironmentVariable:set(key, val)
	if not (key ~= nil) then
		do return end
	end
	_io:write_to_stdout("[jk.env.EnvironmentVariable.set] (EnvironmentVariable.sling:135:3): Not implemented" .. "\n")
end

function jk.env.EnvironmentVariable:unset(key)
	_io:write_to_stdout("[jk.env.EnvironmentVariable.unset] (EnvironmentVariable.sling:141:2): Not implemented" .. "\n")
end

jk.env.TemporaryFile = {}
jk.env.TemporaryFile.__index = jk.env.TemporaryFile
_vm:set_metatable(jk.env.TemporaryFile, {})

function jk.env.TemporaryFile._create()
	local v = _vm:set_metatable({}, jk.env.TemporaryFile)
	return v
end

function jk.env.TemporaryFile:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.env.TemporaryFile'
	self['_isType.jk.env.TemporaryFile'] = true
end

function jk.env.TemporaryFile:_construct0()
	jk.env.TemporaryFile._init(self)
	return self
end

function jk.env.TemporaryFile:create(extension)
	do return _g.jk.env.TemporaryFile:forDirectory(nil, extension) end
end

function jk.env.TemporaryFile:forDirectory(dir, extension)
	local tmpdir = dir
	if tmpdir == nil then
		tmpdir = _g.jk.env.CommonPath:getTemporaryDirectory()
	end
	if tmpdir == nil then
		do return nil end
	end
	do tmpdir:createDirectoryRecursive() end
	if tmpdir:isDirectory() == false then
		do return nil end
	end
	do
		local v = nil
		local n = 0
		local rnd = _g.jk.random.Random._construct0(_g.jk.random.Random._create())
		while n < 100 do
			local id = "_tmp_" .. _g.jk.lang.String:safeString(_g.jk.lang.String:forInteger(_g.jk.time.SystemClock:asSeconds())) .. _g.jk.lang.String:safeString(_g.jk.lang.String:forInteger(_util:convert_to_integer(rnd:nextInt1() % 1000000)))
			if extension == nil or _g.jk.lang.String:getLength(extension) < 1 then
				id = _g.jk.lang.String:safeString(id) .. _g.jk.lang.String:safeString(extension)
			end
			v = tmpdir:entry(id)
			if v:exists() == false then
				do v:touch() end
				do break end
			end
			do n = n + 1 end
		end
		if v ~= nil and v:isFile() == false then
			v = nil
		end
		do return v end
	end
end

function jk.env.TemporaryFile:forString(data)
	if not (data ~= nil) then
		do return nil end
	end
	do
		local v = _g.jk.env.TemporaryFile:create(nil)
		if not (v ~= nil) then
			do return nil end
		end
		if not v:setContentsUTF8(data) then
			do v:remove() end
			do return nil end
		end
		do return v end
	end
end

function jk.env.TemporaryFile:forBuffer(data)
	if not (data ~= nil) then
		do return nil end
	end
	do
		local v = _g.jk.env.TemporaryFile:create(nil)
		if not (v ~= nil) then
			do return nil end
		end
		if not v:setContentsBuffer(data) then
			do v:remove() end
			do return nil end
		end
		do return v end
	end
end

File: .sling/libs/web5/eqela/jk-core/0.36.0/jk.env.js
let JkEnvEnvironmentVariable = function() {};

JkEnvEnvironmentVariable.prototype._t = { "JkEnvEnvironmentVariable" : true };
JkEnvEnvironmentVariable.prototype._tobj = JkEnvEnvironmentVariable;

JkEnvEnvironmentVariable.NEW = function() {
	var v = new JkEnvEnvironmentVariable;
	return v.CTOR_JkEnvEnvironmentVariable();
};

JkEnvEnvironmentVariable.prototype.CTOR_JkEnvEnvironmentVariable = function() {
	return this;
};

JkEnvEnvironmentVariable.getAll = function() {
	console.log("[jk.env.EnvironmentVariable.getAll] (EnvironmentVariable.sling:55:3): Not implemented");
	return null;
};

JkEnvEnvironmentVariable.get = function(key) {
	if(!(key != null)) {
		return null;
	}
	var vars = JkEnvEnvironmentVariable.getAll();
	if(!(vars != null)) {
		return null;
	}
	return JkLangMap.get(vars, key);
};

JkEnvEnvironmentVariable.set = function(key, val) {
	if(!(key != null)) {
		return;
	}
	console.log("[jk.env.EnvironmentVariable.set] (EnvironmentVariable.sling:139:3): Not implemented");
};

JkEnvEnvironmentVariable.unset = function(key) {
	console.log("[jk.env.EnvironmentVariable.unset] (EnvironmentVariable.sling:145:2): Not implemented");
};

JkEnvEnvironmentVariable.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkEnvEnvironmentVariable"] === true;
};

let JkEnvTemporaryDirectory = function() {};

JkEnvTemporaryDirectory.prototype._t = { "JkEnvTemporaryDirectory" : true };
JkEnvTemporaryDirectory.prototype._tobj = JkEnvTemporaryDirectory;

JkEnvTemporaryDirectory.NEW = function() {
	var v = new JkEnvTemporaryDirectory;
	return v.CTOR_JkEnvTemporaryDirectory();
};

JkEnvTemporaryDirectory.prototype.CTOR_JkEnvTemporaryDirectory = function() {
	return this;
};

JkEnvTemporaryDirectory.create = function() {
	return JkEnvTemporaryDirectory.forDirectory(null);
};

JkEnvTemporaryDirectory.forDirectory = function(dir) {
	var ff = JkEnvTemporaryFile.forDirectory(dir, null);
	if(!(ff != null)) {
		return null;
	}
	ff.remove();
	ff.createDirectoryRecursive();
	if(!ff.isDirectory()) {
		return null;
	}
	return ff;
};

JkEnvTemporaryDirectory.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkEnvTemporaryDirectory"] === true;
};

let JkEnvCommonPath = function() {};

JkEnvCommonPath.prototype._t = { "JkEnvCommonPath" : true };
JkEnvCommonPath.prototype._tobj = JkEnvCommonPath;

JkEnvCommonPath.NEW = function() {
	var v = new JkEnvCommonPath;
	return v.CTOR_JkEnvCommonPath();
};

JkEnvCommonPath.prototype.CTOR_JkEnvCommonPath = function() {
	return this;
};

JkEnvCommonPath.getTemporaryDirectory = function() {
	if(JkOsOS.isLinux() || JkOsOS.isMacOS()) {
		return JkFsFile.forPath("/tmp");
	}
	console.log("[jk.env.CommonPath.getTemporaryDirectory] (CommonPath.sling:44:3): Not implemented");
	return JkFsFileInvalid.NEW();
};

JkEnvCommonPath.getHomeDirectory = function() {
	var dirname = JkEnvEnvironmentVariable.get("HOME");
	if(JkLangString.isNotEmpty(dirname)) {
		return JkFsFile.forPath(dirname);
	}
	return null;
};

JkEnvCommonPath.getAppDirectory = function() {
	console.log("[jk.env.CommonPath.getAppDirectory] (CommonPath.sling:125:3): Not implemented");
	return JkFsFileInvalid.NEW();
};

JkEnvCommonPath.getCodeFileForObject = function(oo) {
	if(!(oo != null)) {
		return null;
	}
	console.log("[jk.env.CommonPath.getCodeFileForObject] (CommonPath.sling:152:3): Not implemented");
	return null;
};

JkEnvCommonPath.getProgramFilesDirectory = function() {
	return null;
};

JkEnvCommonPath.getProgramFilesX86Directory = function() {
	return null;
};

JkEnvCommonPath.getApplicationsDirectory = function() {
	return null;
};

JkEnvCommonPath.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkEnvCommonPath"] === true;
};

let JkEnvExternalCommand = function() {};

JkEnvExternalCommand.prototype._t = { "JkEnvExternalCommand" : true };
JkEnvExternalCommand.prototype._tobj = JkEnvExternalCommand;

JkEnvExternalCommand.NEW = function() {
	var v = new JkEnvExternalCommand;
	return v.CTOR_JkEnvExternalCommand();
};

JkEnvExternalCommand.prototype.CTOR_JkEnvExternalCommand = function() {
	return this;
};

JkEnvExternalCommand.findInPath = function(command) {
	if(!(command != null)) {
		return null;
	}
	return null;
};

JkEnvExternalCommand.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkEnvExternalCommand"] === true;
};

let JkEnvTemporaryFile = function() {};

JkEnvTemporaryFile.prototype._t = { "JkEnvTemporaryFile" : true };
JkEnvTemporaryFile.prototype._tobj = JkEnvTemporaryFile;

JkEnvTemporaryFile.NEW = function() {
	var v = new JkEnvTemporaryFile;
	return v.CTOR_JkEnvTemporaryFile();
};

JkEnvTemporaryFile.prototype.CTOR_JkEnvTemporaryFile = function() {
	return this;
};

JkEnvTemporaryFile.create = function(extension) {
	return JkEnvTemporaryFile.forDirectory(null, extension);
};

JkEnvTemporaryFile.forDirectory = function(dir, extension) {
	var tmpdir = dir;
	if(tmpdir == null) {
		tmpdir = JkEnvCommonPath.getTemporaryDirectory();
	}
	if(tmpdir == null) {
		return null;
	}
	tmpdir.createDirectoryRecursive();
	if(tmpdir.isDirectory() == false) {
		return null;
	}
	var v = null;
	var n = 0;
	var rnd = JkRandomRandom.NEW();
	while(n < 100){
		var id = "_tmp_" + (JkLangString.safeString((JkLangString.forInteger((~(~JkTimeSystemClock.asSeconds())))))) + (JkLangString.safeString((JkLangString.forInteger((rnd.nextInt1() % 1000000)))));
		if(extension == null || JkLangString.getLength(extension) < 1) {
			id = (JkLangString.safeString(id)) + (JkLangString.safeString(extension));
		}
		v = tmpdir.entry(id);
		if(v.exists() == false) {
			v.touch();
			break;
		}
		n++;
	}
	if(v != null && v.isFile() == false) {
		v = null;
	}
	return v;
};

JkEnvTemporaryFile.forString = function(data) {
	if(!(data != null)) {
		return null;
	}
	var v = JkEnvTemporaryFile.create(null);
	if(!(v != null)) {
		return null;
	}
	if(!v.setContentsUTF8(data)) {
		v.remove();
		return null;
	}
	return v;
};

JkEnvTemporaryFile.forBuffer = function(data) {
	if(!(data != null)) {
		return null;
	}
	var v = JkEnvTemporaryFile.create(null);
	if(!(v != null)) {
		return null;
	}
	if(!v.setContentsBuffer(data)) {
		v.remove();
		return null;
	}
	return v;
};

JkEnvTemporaryFile.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkEnvTemporaryFile"] === true;
};
